### ISSUE TITLE
`var op = fn()` wrongly gives warning `ObservableStores` with `object of RootObj` type

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#14514]
`var op = fn()` wrongly gives warning `ObservableStores` with `object of RootObj` type

### Example
```nim
# type A* = object # ok
type A* = object of RootObj # bug: spurious warning: Warning: observable stores to 'op' [ObservableStores]
  a: int
proc fn(): A = discard
var op = fn()
```

### Current Output
```
Warning: observable stores to 'op' [ObservableStores]
```

### Expected Output
no warning in this case

### Additional Information
happens also with `nim c koch`
```
devel 1.3.5 177c7f96e93c6b8d9d206cd13f22c0d0d7d4192e
```

**Comment by squattingmonk** [comment]
Did this fix make it into 1.2.2? I'm running 1.2.4 and still having the issue.

**Comment by ghost** [comment]
> Did this fix make it into 1.2.2? I'm running 1.2.4 and still having the issue.

No, 1.2.4 was a hotfix release and only affected nim-gdb, please try 1.2.6
